test/README.md

```diff
@@ -6,87 +6,104 @@ This directory contains tests and testing docs for `Knative Eventing`.
+
+If you have set up a running environment that meets
+[the e2e test environment requirements](#environment-requirements), you can run
+with `test/e2e-tests.sh --run-tests --skip-knative-setup`.
+
+## Running tests with `go test` command
+
+### Running unit tests
+
+You can also use `go test` command to run unit tests:
+
+```shell
+go test -v ./pkg/...
 ```
 
-_Note that to run `presubmit-tests.sh` or `e2e-tests.sh` scripts, you need to
-have a running environment that meets
-[the e2e test environment requirements](#environment-requirements)_
+_By default `go test` will not run [the e2e tests](#running-end-to-end-tests),
+which needs [`-tags=e2e`](#running-end-to-end-tests) to be enabled._
 
-## Running end-to-end tests
+### Running end-to-end tests
 
-To run [the e2e tests](./e2e), you need to have a running environment that meets
+To run [the e2e tests](./e2e) with `go test` command, you need to have a running
+environment that meets
 [the e2e test environment requirements](#environment-requirements), and you need
 to specify the build tag `e2e`.
 
 ```bash
 go test -v -tags=e2e -count=1 ./test/e2e
 ```
 
-### One test case
+#### One test case
 
-To run one e2e test case, e.g. TestKubernetesEvents, use
-[the `-run` flag with `go test`](https://golang.org/cmd/go/#hdr-Testing_flags):
+To run one e2e test case, e.g. TestSingleBinaryEvent, use the
+[-run](https://golang.org/cmd/go/#hdr-Testing_flags) flag with `go test`:
 
 ```bash
-go test -v -tags=e2e -count=1 ./test/e2e -run ^TestKubernetesEvents$
+go test -v -tags=e2e -count=1 ./test/e2e -run ^TestSingleBinaryEvent$
 ```
 
-### Environment requirements
+## Environment requirements
 
 There's couple of things you need to install before running e2e tests locally.
 
-1. `kubetest` installed:
-
-   ```bash
-   go get -u k8s.io/test-infra/kubetest
-   ```
-
-1. [A running `Knative Serving` cluster.]
-1. A docker repo containing [the test images](#test-images)
-
-Simply run the `./test/e2e-tests.sh` script. It will create a GKE cluster,
-install Knative Serving stack with Istio, upload test images to your Docker repo
-and run the end-to-end tests against the Knative Eventing built from source.
-
-If you already have the `*_OVERRIDE` environment variables set, call the script
-with the `--run-tests` argument and it will use the cluster and run the tests.
-Note that this requires you to have Serving and Istio installed and configured
-to your particular configuration setup. Knative Eventing will still built and
-deployed from source.
-
-Otherwise, calling this script without arguments will create a new cluster in
-project `$PROJECT_ID`, start Knative Serving and the eventing system, upload
-test images, run the tests and delete the cluster. In this case, it's required
-that `$KO_DOCKER_REPO` points to a valid writable docker repo.
+1. A running [Knative](https://www.knative.dev/docs/install/) cluster
+2. A docker repo containing [the test images](#test-images)
```

Comment:
In markdown, you can always use "1." and the numbering will be rendered correctly. This way you don't have to worry (or spend effort on) about numbering.

```suggestion
1. A docker repo containing [the test images](#test-images)
```

Reply:
Done.